tests/engine/test_perturb.py

```diff
@@ -37,12 +36,7 @@ def __init__(self, methodName="runTest"):
 
     @classmethod
     def setUpClass(cls):
-        test_path = os.path.realpath("tests/tmp")
```

Comment:
Since you made some effort to refactor this test: Wouldn't it just be easier to use the fixture `tmp_path` here as well?
But it would imply changing to pytest fro unittest.
CoPilot is quite good at it

Reply:
I agree.